### ISSUE TITLE
How to generate deep insight from sample.py

### ISSUE THREAD

**Issue by yeyu2** [microsoft/MM-REACT#9]
Hi there, I successfully run the code in sample.py with tiny changes:
replace AzureOpenAI() to OpenAI() with official model- 
```
llm = OpenAI(model_name="text-davinci-003", temperature=0, max_tokens=MAX_TOKENS)
```
Unfortunately, the result is quite far from the demo app on Huggingface which gives deep insight into a picture, the best it can generate after a conversation is something simple like 
```
AI: This image is likely a scene of a disaster with people standing on a pile of rubble.
```
What else I can do to make a program that similar to the demo app to best utilize the power of MM-REACT? Thanks

**Comment by yeyu2** [comment]
Thanks, Dashesy.
Could you please let me know whether you are going to implement OpenAI() with public gpt-3.5-turbo model someday? As I am a solo developer who has no access to AzureOpenAI at this moment. Thank you.

**Comment by dashesy** [comment]
I got OpenAI() gpt3.5-turbo for my personal usecase but it doesn't support prompting yet. It only offers a limited "conversation" endpoint. Or at least I could not find how to make it work.

**Comment by dashesy** [comment]
Now there is [this branch](https://github.com/microsoft/MM-REACT/tree/ehazar/chat_endpoint) that relies on ChatCompletion client, like the non-azure version of gpt3.5-turbo.
You should be able to pass "chat_completion=True" to OpenAI to use that endpoint

**Comment by yeyu2** [comment]
Really exciting to see this update, thanks!

**Comment by dashesy** [comment]
Addressed by PR #21

**Comment by zairm21** [comment]
> Hi there, I successfully run the code in sample.py with tiny changes: replace AzureOpenAI() to OpenAI() with official model-
> 
> ```
> llm = OpenAI(model_name="text-davinci-003", temperature=0, max_tokens=MAX_TOKENS)
> ```
> 
> Unfortunately, the result is quite far from the demo app on Huggingface which gives deep insight into a picture, the best it can generate after a conversation is something simple like
> 
> ```
> AI: This image is likely a scene of a disaster with people standing on a pile of rubble.
> ```
> 
> What else I can do to make a program that similar to the demo app to best utilize the power of MM-REACT? Thanks

Hi @yeyu2,
I've been trying to test sample.py but it gets stuck with this message: > **Entering new AgentExecutor chain...**
and killed after a while. I tried both davinci-003 and gpt-35-turbo for AzureOpenAI.

I tried passing a text only prompt like : output = chain.conversation("Describe this image") and it generates the following:

1**. There is an image in the input**
**2. The image is new in the conversation**

**<|im_sep|>AI**
**I am sorry, I need an image to describe. Please provide an image.**

**> Finished chain.**

But when I call **output = chain.conversation("https://www.oracle-dba-online.com/sql/weekly_sales_table.png")**, it gets stuck with **> Entering new AgentExecutor chain...**

Can you please let me know how you tested the code without issues? Any specific format to pass the image? I really appreciate that!
